### ISSUE TITLE
The ability to transform a file on extract appears to be missing

### ISSUE THREAD

**Issue by payitforwardnow** [atom/asar#155]
Please forgive me if I've posted this as an issue and it should have been posted elsewhere as a "request". I've noticed that the transform feature only exists when packing files into the archive and it is somewhat limited (my opinion). I have a need to be able to transform a file via compression and/or encryption on a per file basis on-the-fly. By this I mean that based on the type of file being added to the archive I may want to compress the file using a scheme that is best for that type of file (e.g., zip, gzip, custom, etc.). I ALSO want to then have the ability to encrypt/encode the file using a shared or unique encryption key and encryption algorithm of my choice. From what I can tell, on the packing side of things I can only provide a transform that can compress or encrypt but not both do the way the logic is setup (I might be missing something but thats what I see during testing). On the extract side there is no apparent way to run the reverse transform on a file.

I've gone ahead and added these features (on my end) and I'd like to know if there was a way for me to contribute to this project (or electron in general). I'm new to github and I'm learning the way things are done. I'd like to help if I can. 

To recap: I've implemented a way to preform multiple transforms on files (e.g., compress then encrypt, etc) on both the pack and extract side. If anyone is interested in these changes please let me know. 

Thank you in advance for your time ad consideration.
